### ISSUE TITLE
Softmax coarse tiling on reduction axis results in numerical error

### ISSUE THREAD

**Issue by spectre-ns** [torch-spyre/torch-spyre#3726]
### What happened?

The per-block column-sum table (each 128-row block sums to ~1.0 instead of ~0.25, full column 3.9934), max abs err 0.01034 against a max correct magnitude of 0.00319, and the three facts that narrow it to reduction-dim tiling: eager is correct, C-only tiling is correct, and every block is internally self-consistent (which rules out the running-accumulator reading — that would give 1.0 / 0.5 / 0.33 / 0.25).

### What did you expect to happen?

I expected the hints to be observed (with the needed deductions) with numerical correctness or rejected with feedback of the rejection reason to the user via logs.

### How can we reproduce it?

```
import torch
import torch_spyre._inductor.wsr.propagate_named_dims as _pnd
from torch_spyre._inductor import spyre_hint
from torch._inductor.utils import fresh_cache

import torch

def hinted_softmax(x):
    with spyre_hint(num_tiles_per_dim={"R": 4}):
        return torch.softmax(x, dim=0)

DEVICE = torch.device("spyre")
torch.manual_seed(0xAFFE)

# Create input tensor
ROWS, COLS = 512, 1024
x = torch.rand(ROWS, COLS, dtype=torch.float16)

# Compute softmax on the cpu
cpu_result = torch.softmax(x, dim=0)

# Send input tensor to device
x_device = x.to(DEVICE)
_pnd.declare_tensor_dim("R", ROWS)
_pnd.declare_tensor_dim("C", COLS)
_pnd.name_tensor_dims(x_device, ["R", "C"])

# Compute softmax on the device in eager mode and get the result back to the host
eager_result = torch.softmax(x_device, dim=0).cpu()

# Compute softmax on the device in compiled mode and get the result back to the host
with fresh_cache():
    compiled_sm = torch.compile(hinted_softmax)
    compiled_result = compiled_sm(x_device).cpu()

# Print the results and compare them
print(f"CPU result\n{cpu_result}")
print(f"Spyre Eager result\n{eager_result}")
print(f"Spyre Compiled result\n{compiled_result}")
device_delta = torch.abs(eager_result - compiled_result).max()
cpu_delta = torch.abs(compiled_result - cpu_result).max()

print(f"Max delta Compiled Spyre vs. CPU: {cpu_delta}")
print(f"Max delta Compiled Spyre vs. Eager Spyre: {device_delta}")

```

### Any environmental details we need to know?

Reproducible at ca01efe

### Anything else we need to know?

https://claude.ai/code/artifact/913600db-a59f-4663-96fd-190babab49d6?via=auto_preview

### Relevant log output

```Shell
CPU result
tensor([[0.0028, 0.0021, 0.0013,  ..., 0.0028, 0.0020, 0.0025],
        [0.0026, 0.0026, 0.0030,  ..., 0.0013, 0.0011, 0.0014],
        [0.0016, 0.0016, 0.0024,  ..., 0.0014, 0.0024, 0.0025],
        ...,
        [0.0013, 0.0020, 0.0019,  ..., 0.0031, 0.0012, 0.0018],
        [0.0026, 0.0020, 0.0015,  ..., 0.0025, 0.0029, 0.0015],
        [0.0012, 0.0029, 0.0013,  ..., 0.0012, 0.0013, 0.0026]],
       dtype=torch.float16)
Spyre Eager result
tensor([[0.0028, 0.0021, 0.0013,  ..., 0.0028, 0.0020, 0.0025],
        [0.0025, 0.0026, 0.0030,  ..., 0.0013, 0.0011, 0.0014],
        [0.0015, 0.0016, 0.0024,  ..., 0.0014, 0.0024, 0.0025],
        ...,
        [0.0013, 0.0020, 0.0019,  ..., 0.0031, 0.0012, 0.0018],
        [0.0026, 0.0020, 0.0015,  ..., 0.0024, 0.0029, 0.0015],
        [0.0012, 0.0029, 0.0013,  ..., 0.0012, 0.0013, 0.0026]],
       dtype=torch.float16)
Spyre Compiled result
tensor([[0.0112, 0.0083, 0.0055,  ..., 0.0109, 0.0079, 0.0102],
        [0.0101, 0.0105, 0.0123,  ..., 0.0050, 0.0045, 0.0057],
        [0.0062, 0.0065, 0.0099,  ..., 0.0057, 0.0093, 0.0101],
        ...,
        [0.0052, 0.0079, 0.0078,  ..., 0.0131, 0.0047, 0.0072],
        [0.0102, 0.0079, 0.0059,  ..., 0.0105, 0.0117, 0.0058],
        [0.0048, 0.0111, 0.0051,  ..., 0.0052, 0.0051, 0.0104]],
       dtype=torch.float16)
Max delta Compiled Spyre vs. CPU: 0.01033782958984375
Max delta Compiled Spyre vs. Eager Spyre: 0.010345458984375
```

**Comment by spectre-ns** [comment]
Resolved at b440b65
